data_steward/cdr_cleaner/cleaning_rules/remove_aian_participants.py

```diff
@@ -0,0 +1,71 @@
+"""
+Background
+DC-685
+There is a requirement from the NIH that participants who identified as AI/AN in PPI cannot have ANY of their data
+included in released datasets for the Data Browser or Workbench/Cohort Builder. These participants have had their data
+retracted non-programmatically previously. However, since this is expected to be a long-term policy, this should be a
+cleaning rule applied to combined so this can be automated.
+We will remove ALL DATA in combined associated with PIDs who have rows with the following in the observation table:
+observation_source_concept_id = 1586140
+value_source_concept_id = 1586141
+"""
+
+# Project imports
+from cdr_cleaner.cleaning_rules import sandbox_and_remove_pids
+from notebooks import bq
+
+TICKET_NUMBER = 'DC685'
+
+PIDS_QUERY = """
+SELECT person_id
+FROM `{project}.{dataset}.observation`
+WHERE observation_source_concept_id IN(1586140) AND value_source_concept_id IN(1586141)
+"""
+
+
+def get_pids_list(project_id, dataset_id, pids_query):
+    """
+    takes a query based on the cleaning rule and returns a list of person_ids
+    :param project_id: bq name of project_id
+    :param dataset_id: bq name of dataset_id
+    :param pids_query: query that grabs all person_ids based on the cleaning rule
+    :return: list of person_ids
+    """
+
+    pid_list = bq.query(
+        pids_query.format(project=project_id,
+                          dataset=dataset_id))['person_id'].tolist()
+
+    return pid_list
+
+
+def get_queries(project_id, dataset_id):
+    """
+    return a list of queries to remove AIAN participant rows
+    :param project_id: Name of the project
+    :param dataset_id: Name of the dataset where the queries should be run
+    :return: A list of string queries that can be executed to delete AIAN participants and
+    all corresponding rows from the dataset with the associated PID.
+    """
+
+    queries_list = []
+
+    queries_list.append(
```

Comment:
Here and later at line 57, you should be using `queries_list.extend(`.  You are getting a list of query dictionaries from `sandbox_and_remove_pids.get_sandbox_queries` and `sandbox_and_remove_pids.get_remove_pids_queries`.  These functions are returning lists of query dictionaries.  Using `append` here creates a list of lists of query dictionaries.  So, this is returning something with this structure 

```
[
    [
        {query: sandbox query 1, other key:  other value}, 
        {query: sandbox query 2, other key: other value}
    ]
    [
        {query: remove pid query 1, other key:  other value}, 
        {query: remove pid query 2, other key: other value}
    ]
]
```

The clean engine is expecting this data structure a single list of query dictionaries.  Modeled below.

```
[
    {query: sandbox query 1, other key:  other value}, 
    {query: sandbox query 2, other key: other value},
    {query: remove pid query 1, other key:  other value}, 
    {query: remove pid query 2, other key: other value}
]
```

Reply:
Good catch, thank you!